File: app/src/main/java/eu/warble/pjapp/ui/main/student_info/fees/PaymentsListAdapter.java
package eu.warble.pjapp.ui.main.student_info.fees;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.List;

import eu.warble.pjapp.R;
import eu.warble.pjapp.data.model.PlatnosciItem;
import eu.warble.pjapp.data.model.Student;


public class PaymentsListAdapter extends BaseAdapter {
    private List<PlatnosciItem> payments;
    private LayoutInflater inflater;

    public PaymentsListAdapter(@NonNull Context context, Student studentData) {
        inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        this.payments = studentData.getPlatnosci();
    }

    @Nullable
    @Override
    public PlatnosciItem getItem(int position) {
        return payments.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getCount() {
        return payments.size();
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        View view = convertView;
        if (view == null){
            view = inflater.inflate(R.layout.list_item_payments, parent, false);
        }

        PlatnosciItem platnosciItem = getItem(position);
        if (platnosciItem != null) {
            ((TextView) view.findViewById(R.id.payments_list_date))
                    .setText(platnosciItem.getDataWplaty());
            ((TextView) view.findViewById(R.id.payments_list_amount))
                    .setText(String.valueOf(platnosciItem.getKwota()));
            ((TextView) view.findViewById(R.id.payments_list_name))
                    .setText(platnosciItem.getWplacajacy());
            ((TextView) view.findViewById(R.id.payments_list_description))
                    .setText(platnosciItem.getTytulWplaty());
        }

        return view;
    }
}

File: app/src/main/java/eu/warble/pjappkotlin/mvp/studentinfo/fees/adapter/PaymentsListAdapter.kt
package eu.warble.pjappkotlin.mvp.studentinfo.fees.adapter

import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import eu.warble.pjappkotlin.R
import eu.warble.pjappkotlin.data.model.PlatnosciItem
import eu.warble.pjappkotlin.data.model.Student
import eu.warble.pjappkotlin.utils.inflate
import kotlinx.android.synthetic.main.list_item_payments.view.payments_list_amount
import kotlinx.android.synthetic.main.list_item_payments.view.payments_list_date
import kotlinx.android.synthetic.main.list_item_payments.view.payments_list_description
import kotlinx.android.synthetic.main.list_item_payments.view.payments_list_name

class PaymentsListAdapter(
        val studentData: Student
) : RecyclerView.Adapter<PaymentsListAdapter.ViewHolder>() {

    private val payments: List<PlatnosciItem>? = studentData.platnosci

    override fun getItemCount(): Int {
        return payments?.size ?: 0
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(payments?.get(position))
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(parent.inflate(R.layout.list_item_payments))
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val name: TextView = itemView.payments_list_name
        private val amount: TextView = itemView.payments_list_amount
        private val date: TextView = itemView.payments_list_date
        private val description: TextView = itemView.payments_list_description

        fun bind(payment: PlatnosciItem?) = with(itemView) {
            name.text = payment?.wplacajacy
            amount.text = payment?.kwota.toString()
            date.text = payment?.dataWplaty
            description.text = payment?.tytulWplaty
        }
    }

}